libraries/bot-connector/src/test/java/com/microsoft/bot/connector/OAuthTestBase.java

```diff
@@ -0,0 +1,137 @@
+package com.microsoft.bot.connector;
+
+
+import com.microsoft.bot.connector.authentication.AuthenticationConstants;
+import com.microsoft.bot.connector.authentication.MicrosoftAppCredentials;
+import com.microsoft.bot.connector.authentication.OAuthClient;
+import com.microsoft.bot.connector.base.TestBase;
+import com.microsoft.bot.connector.implementation.ConnectorClientImpl;
+import com.microsoft.bot.schema.models.ChannelAccount;
+import com.microsoft.rest.RestClient;
+import com.sun.jndi.toolkit.url.Uri;
+import org.apache.commons.io.FileSystemUtils;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Function;
+
+import static com.ea.async.Async.await;
+import static java.util.concurrent.CompletableFuture.completedFuture;
+
+
+public class OAuthTestBase extends TestBase
+{
+    //private final HttpRecorderMode mode = HttpRecorderMode.Playback;
+
+    protected final String clientId = "adcdf0c4-5ba0-4f46-9823-61ff9c40ca9e";
+    protected final String clientSecret = "6}a4o5D]yhL!jmzI";
+    protected final String userId = "U19KH8EHJ:T03CWQ0QB";
+    protected final String botId = "B21UTEF8S:T03CWQ0QB";
+    protected final static String hostUri = "https://slack.botframework.com";
+
+
+    private String token;
+
+    private ChannelAccount bot;
+    public ChannelAccount getBot() {
+        return this.bot;
+    }
+
+    private ChannelAccount user;
+    public ChannelAccount getUser() {
+        return this.user;
+    }
+
+
+    public OAuthTestBase() throws IOException, ExecutionException, InterruptedException, URISyntaxException {
+        super(RunCondition.BOTH);
+
+        if (true) //mode == HttpRecorderMode.Record)
+        {
```

Comment:
nit
you are a little inconsistent in where you put the { } ...sometimes you are like C# and sometimes like JavaScript